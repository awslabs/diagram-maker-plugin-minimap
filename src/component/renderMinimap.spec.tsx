import { EditorMode } from 'diagram-maker';
import { createMinimap } from '..';

describe('createMinimap', () => {
  const minimap = {
    data:{
      size:{
        height: 200,
        width: 200
      }
    }
  };
  const state = {
    edges: {},
    editor: {
      mode: EditorMode.SELECT
    },
    nodes: {},
    panels: {},
    plugins: { minimap },
    workspace:{
      canvasSize: {
        height: 1600,
        width: 3200
      },
      position: {
        x: 0,
        y: 0
      },
      scale: 1,
      viewContainerSize: {
        height: 800,
        width: 1600
      }
    }
  };
  const renderCallback = jest.fn();
  const destroyCallback = jest.fn();
  it('test createMinimap to match snapshoot', () => {
    const res = createMinimap(state, renderCallback, destroyCallback);
    expect(res).toMatchSnapshot();
  });
});
