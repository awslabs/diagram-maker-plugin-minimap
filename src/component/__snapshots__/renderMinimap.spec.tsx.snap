// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMinimap test createMinimap to match snapshoot 1`] = `
<div>
  <div
    class="dm-minimap-container"
    style="width: 200px; height: 200px;"
  >
    <div
      class="dm-minimap-rectangle"
      data-draggable="true"
      data-event-target="true"
      data-type="DiagramMakerMinimap.Rectangle"
      style="transform: translate3d(0px, 50px, 0); width: 98px; height: 48px;"
    />
    <div
      class="dm-minimap-canvas"
      data-event-target="true"
      data-type="DiagramMakerMinimap.Canvas"
      style="width: 200px; height: 100px; top: 50px; left: 0px;"
    />
  </div>
</div>
`;
